docs/configuration-options.md

```diff
@@ -0,0 +1,117 @@
+
+## Step 1: Laying the Foundation – Establish Benchmark Targets <a name="step-1"></a>
+
+Your journey starts by defining the `benchmark` section within your `build.gradle` file. This section is your playground where you register the compilations you wish to run benchmarks on, within a `targets` subsection.
+
+Targets can be registered in two ways. Either by their name, such as `"jvm"`, which registers its `main` compilation, meaning `register("jvm")` and `register("jvmMain")` will register the same compilation. Alternatively, you can register a source set, for instance, `"jvmTest"` or `"jsBenchmark"`, which will register the corresponding compilation. Here's an illustration using a multiplatform project:
+
+```groovy
+benchmark {
+    targets {
+        register("jvm") 
+        register("js")
+        register("native")
+        register("wasm") // Experimental
+    }
+}
+```
+
+## Step 2: Tailoring the Setup – Create Benchmark Configurations <a name="step-2"></a>
+
+Having laid the groundwork with your targets, the next phase involves creating configurations for your benchmarks. You accomplish this by adding a `configurations` subsection within your `benchmark` block.
+
+The kotlinx benchmark toolkit automatically creates a `main` configuration as a default. However, you can mold this tool to suit your needs by creating additional configurations. These configurations are your control knobs, letting you adjust the parameters of your benchmark profiles. Here's how:
+
+```groovy
+benchmark {
+    configurations {
+        main { 
+            // Configuration parameters for the default profile go here
+        }
+        smoke { 
+            // Create and configure a "smoke" configuration.
+        }       
+    }
+}
+```
+
+## Step 3: Fine-tuning Your Setup – Understanding and Setting Configuration Options <a name="step-3"></a>
+
+Each configuration brings a bundle of options to the table, providing you with the flexibility to meet your specific benchmarking needs. We delve into these options to give you a better understanding and help you make the most of the basic and advanced settings:
+
+**Note:** Many of these configuration options correspond to annotations in the benchmark code. Please be aware that any values provided in the build script will override those defined by annotations in the code. 
+
+### Basic Configuration Options: The Essential Settings <a name="step-3a"></a>
+
+| Option | Description | Default Value | Possible Values |
+| --- | --- | --- | --- |
+| `iterations` | Specifies the number of iterations for measurements. | - | Integer |
+| `warmups` | Specifies the number of iterations for system warming, ensuring accurate measurements. | - | Integer |
+| `iterationTime` | Specifies the duration for each iteration, both measurement and warm-up. | - | Integer |
+| `iterationTimeUnit` | Specifies the unit for `iterationTime`. | Seconds | "ns", "μs", "ms", "s", "m", "h", "d" |
```

Comment:
It would be better to not specify `iterationTimeUnit` and `mode` default values here, as it might be misleading.
Here is why:
As mentioned above, the configuration options override annotation values. So, if `iterationTimeUnit` is not provided, its default value (second) should override the value provided by the corresponding annotation, which is not the case.
In fact, "second" and "throughput" are the default values in annotations.

Reply:
I appreciate your comment and clarification. I have updated the table to reflect this [6d43df4](https://github.com/Kotlin/kotlinx-benchmark/pull/117/commits/6d43df41dcdb8144dbe1412d1a7349a7473d507e).